### PR TITLE
feat(annotations): bring Activity sidebar into view on select

### PR DESCRIPTION
**Pull request by ConradJChan**
- Lifted handling of `updateActiveAnnotation` up from `ActivitySidebar` to `Sidebar` so that it can change the path according to active annotation id changes
- Added logic to force sidebar open when active annotation id transitions to a truthy value

Should handle the scenarios where:
- Another sidebar tab is active, and then selecting or creating an annotation should change the sidebar tab to activity and bring the active annotation feed item into view
- Sidebar is toggled closed, and selecting an annotation or creating an annotation should force the sidebar open to the activity tab and bring the active annotation feed item into view

TODO
 - [x] unit tests
 - [x] update flow types

### CODE REVIEW

File: src/elements/content-sidebar/ActivitySidebar.js [hunk 1]
```diff
@@ -695,6 +671,7 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
             emitAnnotatorActiveChangeEvent,
             file,
             history,
+            getAnnotationsPath,
```
Comment (jstoffan) [inline src/elements/content-sidebar/ActivitySidebar.js:674]: You know what I'm going to say, right.

File: src/elements/content-sidebar/Sidebar.js [hunk 2]
```diff
@@ -103,8 +107,40 @@ class Sidebar extends React.Component<Props, State> {
...
+            prevActiveAnnotationId !== activeAnnotationId &&
+            (isAnnotationsPath || (activeAnnotationId && !isAnnotationsPath))
+        ) {
+            this.updateActiveAnnotation();
+        }
     }
 
+    updateActiveAnnotation = () => {
+        const {
+            annotatorState: { activeAnnotationId } = {},
```
Comment (jstoffan) [inline src/elements/content-sidebar/Sidebar.js:124]: We default the value here, but not in componentDidUpdate. Should this be a `defaultProp` value?

File: src/elements/content-sidebar/Sidebar.js [hunk 3]
```diff
@@ -103,8 +107,40 @@ class Sidebar extends React.Component<Props, State> {
             this.setForcedByLocation();
             this.setState({ isDirty: true });
         }
+
+        // Active annotation id changed. If location is currently an annotation path or
+        // if location is not currently an annotation path but the active annotation id
+        // transitioned from falsy to truthy, update the location accordingly
+        if (
+            prevActiveAnnotationId !== activeAnnotationId &&
+            (isAnnotationsPath || (activeAnnotationId && !isAnnotationsPath))
```
Comment (jstoffan) [inline src/elements/content-sidebar/Sidebar.js:116]: Could pull some of these conditions out into variables to help with readability?

File: src/elements/content-sidebar/Sidebar.js [hunk 4]
```diff
@@ -103,8 +107,40 @@ class Sidebar extends React.Component<Props, State> {
...
+        const currentFileVersionId = getProp(file, 'file_version.id');
+        const fileVersionId = getProp(match, 'params.fileVersionId', currentFileVersionId);
+        const newLocationState = activeAnnotationId ? { ...location.state, open: true } : location.state;
+
+        // Update the location pathname and open state if transitioning to an active annotation id, force the sidebar open
+        history.push({
+            ...location,
+            pathname: getAnnotationsPath(fileVersionId, activeAnnotationId),
+            state: newLocationState,
+        });
```
Comment (mxiao6) [inline src/elements/content-sidebar/Sidebar.js:141]: 👍

File: src/elements/content-sidebar/ActivitySidebar.js (hunk 1 above)
Comment (ConradJChan) [reply src/elements/content-sidebar/ActivitySidebar.js:674]: argh

**Pull request merged**